src/Voron/Data/Graphs/Hnsw.cs

```diff
@@ -479,10 +470,7 @@ public float Distance(Span<byte> vector, int fromIdx, int toIdx)
 
             ref var to = ref GetNodeByIndex(toIdx);
             Span<byte> v2 = to.GetVector(in this);
-            var distance = 1 - TensorPrimitives.CosineSimilarity(
-                MemoryMarshal.Cast<byte, float>(vector),
-                MemoryMarshal.Cast<byte, float>(v2)
-            );
+            var distance = 1 - SimilarityCalc(vector, v2);
```

Comment:
You have `1 - ` in the `HammingDistance`, however.
Better to be consistence and have every `1 - ` etc in the actual functions, not here

Reply:
Done